src/three-domain-secure/component.jsx

```diff
@@ -1,30 +1,54 @@
 /* @flow */
-import { getLogger, getSDKToken } from "@paypal/sdk-client/src";
+import { type LoggerType } from "@krakenjs/beaver-logger/src";
+import { ZalgoPromise } from "@krakenjs/zalgo-promise/src";
+import { create, type ZoidComponent } from "@krakenjs/zoid/src";
 import { FPTI_KEY } from "@paypal/sdk-constants/src";
 
 import { ValidationError } from "../lib";
 
-export const getThreeDomainSecure = (): Function => {
-  const sdkToken = getSDKToken();
-  const ThreeDomainSecureAuth = () => {
-    if (sdkToken) {
-      // eslint-disable-next-line no-console
-      console.log("Three Domain Secure Called");
-      // Make a Zoid component and introduce methods here
-      // onSuccess
-      // onCancel
-      // onClose
-      getLogger()
-        .info("three domain secure v2 invoked")
-        .track({
-          [FPTI_KEY.TRANSITION]: "three_DS_auth_v2",
-        });
-    } else {
-      throw new ValidationError(
-        `script data attribute sdk-client-token is required but was not passed`
-      );
-    }
-  };
-
-  return ThreeDomainSecureAuth;
+type SdkConfig = {|
+  sdkToken: ?string,
```

Comment:
nit: We can probably remove the `?` here. It is required to be a string

Reply:
Agreed!

Reply:
Removing this creates a flow error. I believe that it is possible for null or undefined to be returned by `getSDKToken` which is why it should be optional